core/src/main/kotlin/no/strooped/view/screen/LobbyScreen.kt

```diff
@@ -0,0 +1,145 @@
+class LobbyScreen(val controller: StroopedController) : Screen {
+    var ui: Stage = Stage(ScreenViewport())
+
+    var cam: OrthographicCamera = OrthographicCamera()
+    var mouse: Vector3 = Vector3()
+    var background: Texture = Texture("white.jpg")
+    var exitBtn: Texture = Texture("exit_game_button.png")
+    var loadAnim: Array<Texture?>
+
+    var elapsedTime = 0f
+    private lateinit var batch: SpriteBatch
+
+    private var iterat = 0
+    private val numberOfFrames = 4
+
+    private val backPosX = cam.position.x - cam.viewportWidth / 2f
+    private val backPosY = 0f
+
+    private val exitBtnPosX = (Gdx.graphics.width.toFloat() - TextureSizes.exitGameButtonWidth()) * 0.5f
+    private val exitBtnPosY = Gdx.graphics.height.toFloat() * 0.1f
+
+    private val loadPosX = (Gdx.graphics.width.toFloat() - TextureSizes.loadWidth()) * 0.5f
+    private val loadPosY = (Gdx.graphics.height.toFloat() - TextureSizes.loadHeight()) * 0.35f
+    private val refreshInterval = 0.1f
+
+    init {
+        cam.setToOrtho(false, Gdx.graphics.width.toFloat(), Gdx.graphics.height.toFloat())
+        batch = SpriteBatch()
+
+        loadAnim = arrayOfNulls(numberOfFrames)
+        for (i in 0 until numberOfFrames) {
+            val nameOfImage = "load$i.jpg"
+            loadAnim[i] = Texture(nameOfImage)
+        }
+        val messsage = "Hey " + controller.username + "!\nPlease wait for the lobby-master to start " +
+            "the game.\nWe made a spinning circle you can enjoy while you wait."
+        val label = Label(messsage, Label.LabelStyle(BitmapFont(Gdx.files.internal("chunkfive.fnt")), Color.FIREBRICK))
+        label.setFontScale(FONT_SIZE_WELCOME_TEXT)
+        val labelWidth = Gdx.graphics.width * 0.8f
+        label.width = labelWidth
+        val labelPosX = (Gdx.graphics.width - label.width) * 0.5f
+        val labelPosY = Gdx.graphics.height * 0.7f
+        label.setPosition(labelPosX, labelPosY)
+        label.setWrap(true) // fit inside the label
+        label.setAlignment(1) // center the text
+        ui.addActor(label)
+
+        /*val toastFactory: Toast.ToastFactory = Builder()
+            .font(attr.font)
+            .build()*/
+        // https://github.com/wentsa/Toast-LibGDX - for toasts
+    }
+
+    override fun hide() {}
+
+    override fun show() {
+        batch = SpriteBatch()
+    }
+
+    override fun render(delta: Float) {
+        handleInput()
+        elapsedTime += Gdx.graphics.deltaTime
+        batch.projectionMatrix = cam.combined
+        batch.begin()
+        batch.draw(
+            background,
+            backPosX,
+            backPosY,
+            Gdx.graphics.width.toFloat(),
+            Gdx.graphics.height.toFloat()
+        )
+        batch.draw(
+            exitBtn,
+            exitBtnPosX,
+            exitBtnPosY,
+            TextureSizes.exitGameButtonWidth(),
+            TextureSizes.exitGameButtonHeight()
+        )
+        if (elapsedTime > refreshInterval) {
```

Comment:
I don’t think that this logic should be here: As mentioned over should this logic be placed inside a custom Spinner element, to hide the underlying logic.

Then when adding it to the `batch` can you have a custom `draw` method on the spinner, which accepts deltaTime as argument.

Reply:
Can try and implement this

Reply:
Done